drivers/storage/portworx/util/util.go

```diff
@@ -1100,6 +1100,29 @@ func IsCCMGoSupported(pxVersion *version.Version) bool {
 	return pxVersion.GreaterThanOrEqual(MinimumPxVersionCCMGO)
 }
 
+// ValidateTelemetry validates if telemetry is enabled correctly
+func ValidateTelemetry(cluster *corev1.StorageCluster) error {
+	if !IsTelemetryEnabled(cluster.Spec) {
+		return nil
+	}
+
+	pxVersion := GetPortworxVersion(cluster)
+	if pxVersion.LessThan(MinimumPxVersionCCM) {
+		// PX version is lower than 2.8
+		return fmt.Errorf("telemetry is not supported on Portworx version: %s", pxVersion)
+	} else if IsCCMGoSupported(pxVersion) {
+		if proxy := GetPxProxyEnvVarValue(cluster); proxy != "" {
```

Comment:
I thought we will just disable it when https and/or username password are specified in the URL. we should not disable it if http is used.

Reply:
you're right, i missed the https part in the comment. for http we should not disable it.

Reply:
actually I had a discussion with Bhavana and Prashant, they said we should also disable telemetry with http proxy too. could you confirm with Harsh?